test/ibm/runtime/test_runtime.py

```diff
@@ -622,6 +622,14 @@ def test_jobs_filter_by_program_id(self):
         self.assertEqual(program_id, rjobs[0].program_id)
         self.assertEqual(1, len(rjobs))
 
+    def test_jobs_filter_by_provider(self):
+        """Test retrieving jobs by provider."""
+        program_id = self._upload_program()
+        job = self._run_program(program_id=program_id)
+        job.wait_for_final_state()
+        with self.assertRaises(IBMInputValueError):
+            self.runtime.jobs(hub="defaultHub")
```

Comment:
For the positive test case, you can pass some random hub, group and project to `_run_program` and call `_run_program` twice with two different hgps and then retrieve the job with one of the hgps. 

Implementation details:
In BaseFakeRuntimeClient class you might have to add a method like `set_hgp(h, g, p)` and set hgp as global variables in that class (_hub, _group, _project), just like `_job_classes` or `_final_status`. Call `set_hgp(h, g, p)` in `_run_program` just like we call `set_final_status` or `set_job_status`. And then in `BaseFakeRuntimeClient.program_run`, if `_hub`, `_group` and `_project` global variables are available then pass those to `job_cls`, else pass credentials.hub, credentials.group, credentials.project just like today.